packages/ruleset/src/functions/operation-id-naming-convention.js

```diff
@@ -74,51 +82,100 @@ function operationIdNamingConvention(resolvedSpec) {
+// delete | /albums/{id}                | Y             | 1            | delete       | delete_album
+//        | /albums/{id}/genre          | N             | 1            | unset        | unset_album_genre
+//        | /albums/{id}/producer/{pid} | Y             | 2+           | remove       | remove_album_producer
+//
+// Please try to keep this table in sync with the API Handbook guidance and the logic in the function below.
+//
 function operationIdPassedConventionCheck(
   opKey,
   operationId,
   allPathOperations,
-  pathEndsWithParam
+  pathEndsWithParam,
+  numParamRefs
 ) {
-  // Only consider paths for which
-  // - paths that do not end with param have a GET and POST operation
-  // - paths that end with param have a GET, DELETE, POST, PUT or PATCH.
-
   const verbs = [];
 
-  if (!pathEndsWithParam) {
-    // operationId for GET should starts with "list"
-    if (opKey === 'get') {
-      verbs.push('list');
-    }
-    // operationId for POST should starts with "create" or "add"
-    else if (opKey === 'post') {
-      verbs.push('add');
+  switch (opKey) {
+    case 'get':
+      if (!pathEndsWithParam) {
+        verbs.push('list');
+      } else {
+        verbs.push('get');
+      }
+      if (numParamRefs > 1) {
+        verbs.push('check');
+      }
+      break;
+
+    case 'post':
+      // Need this more lenient logic for VPC.
       verbs.push('create');
-    }
-  } else {
-    // operationId for GET should starts with "get"
-    if (opKey === 'get') {
-      verbs.push('get');
-    }
-    // operationId for DELETE should starts with "delete"
-    else if (opKey === 'delete') {
-      verbs.push('delete');
-    }
-    // operationId for PATCH should starts with "update"
-    else if (opKey === 'patch') {
-      verbs.push('update');
-    }
-    // If PATCH operation doesn't exist for path, POST operationId should start with "update"
-    else if (opKey === 'post') {
-      if (!allPathOperations.includes('patch')) {
+      if (numParamRefs > 0 && !allPathOperations.includes('patch')) {
         verbs.push('update');
       }
-    }
-    // operationId for PUT should starts with "replace"
-    else if (opKey === 'put') {
-      verbs.push('replace');
-    }
+
+      // A normal POST with no param refs should use 'create'.
+      // if (numParamRefs === 0) {
+      //   verbs.push('create');
+      // } else if (!allPathOperations.includes('patch')) {
+      //   // If the path has param refs and there is no corresponding PATCH operation,
+      //   // then the operationId should start with 'update'.
+      //   verbs.push('update');
+      // }
+      break;
+
+    case 'patch':
+      verbs.push('update');
+      break;
+
+    case 'put':
+      if (!pathEndsWithParam) {
+        verbs.push('replace');
+        if (numParamRefs > 0) {
+          verbs.push('set');
+        }
+      } else {
+        verbs.push(numParamRefs > 1 ? 'add' : 'replace');
+      }
+      break;
+
+    case 'delete':
```

Comment:
The handling of DELETE is also a little more lenient than what is specified in the API Handbook.
This was done to avoid new warnings in the vpc api.
Need to discuss this as well.